source/github-helpers/toast.tsx

```diff
@@ -15,7 +15,7 @@ export function ToastSpinner(): JSX.Element {
 
 type Task = () => Promise<unknown>;
 export default async function showToast<TTask extends Task>(
-	task: TTask,
+	task: ((progress: (message: string) => void) => Promise<void>) | TTask,
```

Comment:
Instead of changing it here, `TTask` should be updated. Probably:

```ts
type ProgressCallback = (message: string) => void;
type Task = (progress: ProgressCallback) => Promise<void>;
```

Reply:
Lint fails.
on 
```js
const result = await task((message: string): void => {
```

Reply:
Let's wait for TypeScript help 😃 

I think that line doesn't need types though because they're inferred from `task()`  itself:

```ts
const result = await task(message => {
```

Reply:
Yes, it doesn't need. Where is TypeScript help?